### DISCUSSION TITLE
RPC...

### DISCUSSION THREAD

**Discussion by progman1** [ocaml/dune#6188] [Q&A]
with regard to #6159 the docs still say:
"Definitions of available requests, notifications, and their associated types"

the mli file pointed to says 
"Implementation of the protocol used by dune rpc. Independent of IO and any  specific rpc requests. "

this is a contradiction, no?
the response to #6146 was that the mli was the best available document
which probably means excepting the source code.
so I've been rummaging through the code and made some minor progress with src/dune_rpc/dune_rpc_impl.
is this the 'usable client' mentioned in readthedocs?
I'm having a hard time extricating it from the build context of dune as a whole. I am also looking at test/expect-test/dune_rpc/dune_rpc_e2e but seemingly dependent on the former.
perhaps struggling is good for the soul but in this moment of weakness I will ask: is there a more 'useable' standalone client implementation
available, a demo/skeleton?

after all,  the mli says:
" The protocol described here is stable and is relied
    on by 3rd party clients."
3rd parties = JST teams only?

**Comment by emillon** [comment]
Hi,
The best answer we can give is that the RPC system is not complete yet and not ready for use.
Please do not open new issues. You can post questions on #6146 even if it's marked as completed.
Thanks.

**Comment by rgrinberg** [comment]
If you have general questions, the discussions tab is open and is widely monitored by the team.

**Comment by rgrinberg** [comment]
We have an Lwt client instantiated. This is the most usable client available to end users. Look at the corresponding tests for this library for an example on how to use this library.
